### PR TITLE
Core/Spells: Fixed setting and unsetting of m_spellModTakingSpell

### PR DESCRIPTION
**Pull request by ghost**
**Target branch(es):** 3.3.5

- [x] 3.3.5

**Tests performed:** Compiles

Fixed setting and unsetting of m_spellModTakingSpell, a problem when a different spell tries to overwrite existing ModTakingSpell should no longer happen. Any feedback is welcomed.

### CODE REVIEW

**Comment by shenhuyong** [comment]
Table 'spell_threat' from db not working after I merged this pr.

**Comment by xinef1** [comment]
Totally not related

**Comment by Viste** [comment]
@shenhuyong not true for me and in code no changes to do that

File: src/server/game/Spells/Spell.cpp [hunk 1]
```diff
@@ -7129,13 +7129,7 @@ void Spell::DoAllEffectOnLaunchTarget(TargetInfo& targetInfo, float* multiplier)
         }
     }
 
-    if (Player* modOwner = m_caster->GetSpellModOwner())
-        modOwner->SetSpellModTakingSpell(this, true);
```
Comment (ariel-) [inline src/server/game/Spells/Spell.cpp]: If you remove this, SPELLMOD_CRIT won't register mod in auras, preventing charges dropping. If this is intended then 4829cca9fb81b27cd9a057355ee7b03ff71e0a14 shall be reverted

**Pull request merged**